### ISSUE TITLE
Module augmentation with enum declaration merging produces confusing mixed results

### ISSUE THREAD

**Issue by peterflynn** [microsoft/TypeScript#44834]
# Bug Report

**TL;DR** – when declaration merging an enum across modules, the _types_ are merged but _not_ the generated code that sets up runtime values for the enum members. This gives the appearance that all the enum members exist (type checking, code completion in VSCode, etc.) when actually some of them are undefined.

### 🔎 Search Terms

module augmentation, declaration merging, enum merging, enum value undefined, ambient module enum

### 🕗 Version & Regression Information

Reproduced in both 4.2.3 & 4.3.4 - no reason to believe it's a regression.

- This is the behavior in every version I tried, and I reviewed the ~FAQ~ docs for entries about: declaration merging

### ⏯ Playground Link

_I'm not sure how to make a multi-module Playground – is it possible?  I see some in the canned Examples but it seems to use a hardcoded list of predefined modules only?_

### 💻 Code

**index.ts**
```ts
// Example 1: "Foo" enum - Declaration merging within a single module
export enum Foo {
    one = "one"
}
export enum Foo {
    two = "two"
}

// Example 2: "Bar" enum - Declaration in mod1 plus module augmentation in mod2
import { Bar } from "./mod1";


// *Types* are merged fine in either case
const f1: Foo = Foo.one;
const f2: Foo = Foo.two;
// const f3: Foo = Foo.three; // error - not a member of Foo

const b1: Bar = Bar.abc;
const b2: Bar = Bar.xyz;
// const b3: Bar = Bar.def; // error - not a member of Bar


// *Values* are not merged though
console.log(f1, f2);  // "one" "two"
console.log(b1, b2);  // "abc" "undefined" <-- !!!
```

**mod1.ts**
```ts
export enum Bar {
    abc = "abc"
}
```

**mod2.ts**
```ts
// (Ignore - mod2 must export *something* to be considered a module at all)
export function testFn(): void {
    console.log("This is a test!");
}


// Module augmentation of mod1
declare module "./mod1" {

    // Declaration merging with mod1's definition of Bar
    export enum Bar {
        xyz = "xyz"
    }
}
```

### 🙁 Actual behavior

* Type checking tells you the `Bar` enum has two members: `abc` and `xyz`.
* But at runtime, `xyz` is undefined — only `abc` has the value that was defined in the enum declaration. (As seen in the 2nd `console.log()`).

Contrast this to the `Foo` enum which uses declaration merging _within_ a single file — that one works as expected.

### 🙂 Expected behavior

The two should be consistent, so _either_...

* Type checking tells you the `Bar` enum has two members: `abc` and `xyz`.
* At runtime both have the value that was defined in the enum declaration (neither is undefined)

_Or..._

* Type checking tells you the `Bar` enum has only one member, `abc`

_Or..._

* Compiler error: "Declaration merging enums is only supported if all declarations are in the same module file"
* Or perhaps: "Enums cannot be declared in an ambient module" - at all. (Module augmentation only permits "ambient" declarations, right? Should it _always_ be an error for ambient modules to contain any constructs that generate code?)

**Comment by DanielRosenwasser** [comment]
A module augmentation is fully ambient (anything with a `declare` modifier is), and ambient declarations

1. Have no emit
2. Are purely to inform TypeScript of the true behavior at runtime

So really you're telling TypeScript *someone* is adding another member to this `enum`. Others have to find a way to make that true at runtime.
